### PR TITLE
Changes to lead to OpenSSL 3

### PR DESCRIPTION
**Pull request by ryanskeith**
- Adding in anything that does not have an upper bound on openssl
- Skipping _anaconda_depends as it is a meta package.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -893,6 +893,11 @@ def patch_record_in_place(fn, record, subdir):
     # openssl >=1.1.1,<1.1.2.0a0 -> >=1.1.1a,<1.1.2a
     replace_dep(depends, "openssl >=1.1.1,<1.1.2.0a0", "openssl >=1.1.1a,<1.1.2a")
 
+    # openssl3 preventive measures
+    replace_dep(depends, "openssl !=1.1.1e", "openssl !=1.1.1e,<1.1.2a")
```
Comment (cbouss) [inline main.py:897]:
```suggestion
    replace_dep(depends, "openssl !=1.1.1e", "openssl !=1.1.1e,<1.1.2a")
    replace_dep(constrains, "openssl >=1.1.1k", "openssl >=1.1.1k,<1.1.2a")
```

navigator-updater has some open constraints.
e.g.:
```
    "navigator-updater-0.3.0-py38h06a4308_0.conda": {
      "build": "py38h06a4308_0",
      "build_number": 0,
      "constrains": [
        "pyopenssl >=17.5.0",
        "cryptography >=3.3.2",
        "libxml2 >=2.9.10",
        "expat >=2.2.8",
        "openssl >=1.1.1k",
        "urllib3 >=1.26.4"
      ],
      "depends": [
        "conda",
        "psutil >=5.6.5",
        "pyqt >=5.6,<6.0a0",
        "python >=3.8,<3.9.0a0",
        "pyyaml >=5.4",
        "qtpy >=1.4.1",
        "setuptools",
        "ujson >=4.0.1"
      ],
```

**Review by cbouss — COMMENTED** [review]
I went quickly over a repodata (no script or anything).
The change looks good to me, but you should also verify the constrains section.

**Comment by ryanskeith** [comment]
@cbouss , I did find a few dependencies in contrains that needed to be address.  Thank you for the thoughtful feedback.

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**